Sage/carbon
# feat(sidebar): add padding props for customising content container (FE-4458)

resolves #4494

### Proposed behaviour

- Added styled system padding props to `Sidebar`, which are applied to the sidebar content container. Updated PropTypes, types in `d.ts` file and storybook prop table
- Define default content padding using design tokens:
- Added new story to demonstrate that custom padding can be added
- Minor refactors:
   - remove `defaultProps`
   - rename inner style component to fit convention
   - modify prop types of `size` and `position` props - so all of their possible values are displayed clearly in storybook, as opposed to `SIDEBAR_SIZES` and `SIDEBAR_ALIGNMENTS` 



### Current behaviour

- Cannot pass custom padding props to `Sidebar`
- Padding of content is not defined using design tokens

### Checklist




#### QA


### Additional context

The content padding styles have been updated with design tokens to match the DS design: 

![Screenshot 2022-08-03 at 14 15 36](https://user-images.githubusercontent.com/18368713/182631175-1e0f2027-6c96-44e4-97a6-c06ba56a5a79.png)


### Testing instructions

- Check new story `custom-padding-around-content` to see new behaviour.
- Ensure no regressions have been introduced as a result of this change.
- Double check for any ambiguity in storybook docs for `Sidebar` as a result of this change.
